### PR TITLE
LPD-26543 Pass expando-based custom fields and object custom fields inside the system object definition DTO via object validation payload

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/util/ObjectValidationRuleThreadLocal.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		Map<Long, Set<Long>> objectEntryIdsMap =
+			_objectEntryIdsMapThreadLocal.get();
+
+		Set<Long> objectEntryIds = objectEntryIdsMap.computeIfAbsent(
+			objectValidationRuleId, key -> new HashSet<>());
+
+		objectEntryIds.add(objectEntryId);
+	}
+
+	public static boolean isSkipObjectValidationRuleExecution(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/util/ObjectValidationRuleThreadLocal.java:32]: Why the logic of this method is different from [ObjectActionThreadLocal.isSkipObjectActionExecution](https://github.com/liferay/liferay-portal/blob/e15f84ee11abac54ee86da392eb0c8bcc6442401/modules/apps/object/object-api/src/main/java/com/liferay/object/action/util/ObjectActionThreadLocal.java#L56-L58)? Carol said that she copied and pasted this code from ObjectActionThreadLocal class

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 2]
```diff
@@ -3375,6 +3385,24 @@ public void testUpdateObjectEntry() throws Exception {
 	public void testUpdateObjectEntryWithJavaDelegateObjectValidationRule()
 		throws Exception {
 
+		BiFunction<Map<String, Object>, String, String> biFunction =
+			(inputObjects, sourceName) -> {
+				if (inputObjects.containsKey(sourceName) &&
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:3390]: Is this condition working properly? Checking the old logic, it doesn't seem to be correct.

**Review by guilhermedcamacho — APPROVED** [review]

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:test:object

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/3423

**Pull request closed without merging**